### ISSUE TITLE
[Bug]: <x-date multi field issues?

### ISSUE THREAD

**Issue by Dean1292** [tasteui/tasteui#374]
### What Happened

Hi there,

I am having issues with multi field options for <x-date>, I will attach a video of the issue I am having.
[Screen Recording 2024-03-21 at 18.27.09.mov.zip](https://github.com/tallstackui/tallstackui/files/14705542/Screen.Recording.2024-03-21.at.18.27.09.mov.zip)

So, I wasn't sure if this was due to the code used to add / remove or an issue with maybe <x-date itself as I know it's something that has been newly added but I didn't think that would affect it's use in a multi field array in any way. I did look at another issue submitted about a todolist bug test that seemed to be having similar issues but the fix applied there was due to the code, attempting that fix here did not seem to help

### How to Reproduce

So, here is the relevant parts of my component:

```php
public $bpAttributes = [];

public function addField($attributeId)
    {
        $this->bpAttributes[$attributeId]['value'] = collect( $this->bpAttributes[$attributeId]['value'])->filter()->toArray();
        if ($this->bpAttributes[$attributeId]['type'] === 'datetime') {
            $this->bpAttributes[$attributeId]['value'][] = [
                'date' => '',
                'time' => '',
            ];
        } else {
            $this->bpAttributes[$attributeId]['value'][] = '';
        }
    }

    public function removeField($attributeId, $key)
    {
        unset($this->bpAttributes[$attributeId]['value'][$key]);

        $this->bpAttributes[$attributeId]['value'] = array_values($this->bpAttributes[$attributeId]['value']);

        $this->bpAttributes[$attributeId]['value'] = collect( $this->bpAttributes[$attributeId]['value'])->filter()->toArray();

    }
``` 

```blade
<div class="m-t-md">
    <table class="min-w-full divide-y divide-gray-300">
        <thead class="bg-gray-200">
        <tr>
            <th class="uppercase text-xs font-semibold border-l border-gray-400 text-left pl-2 py-1">Attributes</th>
            <th class="uppercase text-xs font-semibold border-l border-gray-400 text-right pl-2 py-1 pr-5">Actions</th>
        </tr>
        </thead>
        <tbody>
        @foreach($this->existingAttributes as $key => $attribute)
            <tr class="border-b">
                <td class="px-6 capitalize" style="overflow: visible!important;">
                    <p class="font-bold">{{ $attribute->name }}</p>
                    @foreach($bpAttributes[$attribute->id]['value'] as $newKey => $value)
                        @if($bpAttributes[$attribute->id]['is_multiple'])
                            @if($bpAttributes[$attribute->id]['type'] === 'datetime')
                                <div wire:key="{{$newKey}}-{{$attribute->id}}" class="flex items-center"> <!-- Added 'items-center' to align contents vertically -->
                                    <div class="w-1/2 mr-2" wire:key="{{$newKey}}-{{$attribute->id}}">
                                        <x-date wire:key="bpAttributes.{{ $attribute->id }}.value.{{$newKey}}.date" wire:model.live="bpAttributes.{{ $attribute->id }}.value.{{$newKey}}.date" required/>
                                    </div>
                                    <div class="w-1/2" wire:key="{{$newKey}}-{{$attribute->id}}">
                                        <x-time wire:key="bpAttributes.{{ $attribute->id }}.value.{{$newKey}}.time" :step-hour="1" :step-minute="10" wire:model.live="bpAttributes.{{ $attribute->id }}.value.{{$newKey}}.time" required/>
                                    </div>
                                    <div class="inline-block ml-2" wire:key="{{$newKey}}-{{$attribute->id}}"> <!-- Move the remove button inside the loop -->
                                        <button style="color: red; position: relative;" type="button" title="Remove field" class="close" wire:click="removeField({{ $attribute->id }}, {{ $newKey }})">&times;</button>
                                    </div>
                                </div>
                            @endif
                        @endif
                    @endforeach
                    <button type="button" class="mt-2 inline-flex items-center rounded border border-transparent bg-blue-600 px-2 py-1 text-xs font-medium text-white shadow-sm hover:bg-blue-700 focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:ring-offset-2 disabled:bg-gray-200 disabled:text-gray-400 disabled:cursor-not-allowed" wire:click="addField({{ $attribute->id }})">Add Value</button>
                </td>
            </tr>
        @endforeach
        </tbody>
    </table>
</div>
``` 

### Laravel Version

11.0.5

### Livewire Version

3.3.5

### TallStackUI Version

1.20.0

### PHP Version

8.3.3

### Operation System

macOS

**Comment by Dean1292** [comment]
Just for further update, when I change it to be normal input fields, it works as expected however when I use the date and time fields and remove one I get:

tallstackui-CicPENmP.js:6 Uncaught TypeError: Cannot read properties of null (reading 'split')
    at Proxy.hydrate (tallstackui-CicPENmP.js:6:76495)
    at tallstackui-CicPENmP.js:6:76425
    at livewire.esm.js:1387:13

**Comment by Dean1292** [comment]
I also am getting the same with <x-select> has anything changed that would affect all of these? Otherwise it does lead me to believe it could be the setup of it overall

**Comment by devajmeireles** [comment]
Please keep posting as many details as possible and thank you for every description so far. From tomorrow I will start an analysis on this.

**Comment by Dean1292** [comment]
Thank you! So the main details are, when I change to x-input the multi fields appear fine, I can remove and add with no issues at all and the data is in the boxes correctly. When I switch too <x-select or try the date or time inputs that's when I get the issue. No other update other than that, just thought I would include that as a brief overview of what is happening

**Comment by devajmeireles** [comment]
@Dean1292 ,

I was able to reproduce the issue and fix the issue. I'm preparing a new release with this fix.

![CleanShot 2024-03-22 at 15 55 04](https://github.com/tallstackui/tallstackui/assets/60591772/28a057d9-a4f1-458e-af96-b7a49f53d141)

**Comment by devajmeireles** [comment]
Hey, @Dean1292 !

We just released v1.21 with the fix for this issue. Please, upgrade your project dependencies and try it again.

Please, feel free to open a new issue whenever you need to.
